pkg/operator/ceph/csi/operator_config.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8scsiv1 "k8s.io/api/storage/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const opConfigCRName = "ceph-csi-operator-config"
+
+func (r *ReconcileCSI) CreateOrUpdateOpConfig(cluster cephv1.CephCluster) error {
+	logger.Info("Creating ceph-CSI operator config CR")
+
+	opConfig := &csiopv1a1.OperatorConfig{}
+	opConfig.Name = opConfigCRName
+	opConfig.Namespace = r.opConfig.OperatorNamespace
+	spec := *r.generateCSIOpConfigSpec(cluster, opConfig)
+	opConfig.Spec = spec
+
+	err := r.client.Create(r.opManagerContext, opConfig)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "failed to create CSI-operator operator config CR %q", opConfig.Name)
+		}
+		logger.Infof("CSI-operator operator config cr %q already exist, updating...", opConfig.Name)
+		err := r.client.Get(r.opManagerContext, types.NamespacedName{Name: opConfigCRName, Namespace: r.opConfig.OperatorNamespace}, opConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get CSI-operator operator config CR %q", opConfigCRName)
+		}
+
+		opConfig.Spec = spec
+
+		err = r.client.Update(r.opManagerContext, opConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update CSI-operator operator config CR %q", opConfig.Name)
+		}
+	}
+
+	logger.Info("Successfully updated ceph-CSI operator config CR")
+
+	return nil
+}
+
+func (r *ReconcileCSI) generateCSIOpConfigSpec(cluster cephv1.CephCluster, opConfig *csiopv1a1.OperatorConfig) *csiopv1a1.OperatorConfigSpec {
```

Comment:
@subhamkrai where are we creating 3 driver CR's?

Reply:
@Madhu-1 for the driver CR's @sp98 is working on those to fasten the rook integration.